controllers/securitypolicy_controller.go

```diff
@@ -248,46 +243,88 @@ func (r *SecurityPolicyReconciler) create(ctx context.Context, policy *tykv1.Sec
 		}
 	}
 
-	err = r.updateLinkedAPI(ctx, policy, func(ads *tykv1.ApiDefinitionStatus, s model.Target) {
-		ads.LinkedByPolicies = addTarget(ads.LinkedByPolicies, s)
-	})
-
+	err = r.updateStatusOfLinkedAPIs(ctx, policy, false)
 	if err != nil {
 		r.Log.Error(err, "failed to update linkedAPI status")
 	}
 
-	r.Log.Info("Successful created Policy")
+	r.Log.Info("Successfully created Policy")
 
 	policy.Spec.MID = spec.MID
+
+	return r.updatePolicyStatus(ctx, policy)
+}
+
+// updatePolicyStatus updates the status of the policy.
+func (r *SecurityPolicyReconciler) updatePolicyStatus(ctx context.Context, policy *tykv1.SecurityPolicy) error {
+	r.Log.Info("Updating policy status")
+
 	policy.Status.PolID = policy.Spec.MID
 
+	if policy.Spec.AccessRightsArray != nil && len(policy.Spec.AccessRightsArray) > 0 {
+		policy.Status.LinkedAPIs = make([]model.Target, 0)
+	} else {
+		policy.Status.LinkedAPIs = nil
+	}
+
+	for _, v := range policy.Spec.AccessRightsArray {
+		target := model.Target{Name: v.Name, Namespace: v.Namespace}
```

Comment:
what if `v.Namespace` is an empty string? Should we assign it to `policy.Namespace` if empty?

Reply:
I don't think that's good idea. If namespace is invalid, then it should not even call this function.
It should fail before.